src/server/chromium/videoRecorder.ts

```diff
@@ -124,8 +126,8 @@ export class VideoRecorder {
 
     if (this._lastWriteTimestamp) {
       const durationSec = (Date.now() - this._lastWriteTimestamp) / 1000;
-      if (durationSec > 1 / fps)
-        this.writeFrame(this._lastFrameBuffer!, this._lastFrameTimestamp + durationSec);
+      if (!this._lastWritePromise || durationSec > 1 / fps)
+        this._flushLastFrame(durationSec).catch(e => this._progress.log('Error while writing frame: ' + e));
```

Comment:
Shouldn't you await it?

Reply:
No, we chain them and wait for the tail in stop().

Reply:
Can we chain them explicitly?